keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,147 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trainer for Word Piece Tokenizer."""
+
+import tensorflow as tf
+from tensorflow_text.tools.wordpiece_vocab import (
+    wordpiece_tokenizer_learner_lib as learner,
+)
+
+from keras_nlp.tokenizers.word_piece_tokenizer import pretokenize
+
+
+def compute_word_piece_vocabulary(
+    data,
+    vocabulary_size,
+    vocabulary_output_file=None,
+    lowercase=True,
+    strip_accents=True,
+    split=True,
+    suffix_indicator="##",
+    reserved_tokens=["[PAD]", "[CLS]", "[SEP]", "[UNK]", "[MASK]"],
+):
+    r"""A utility to train a Word Piece vocabulary.
+
+    Trains a Word Piece vocabulary from an input dataset or a list of filenames.
+
+    For custom data loading and pretokenization (`split=False`), the input
+    `data` should be a `tf.data.Dataset`. If `data` is a list of filenames,
+    the file format is required to be plain text files, and the text would be
+    read in line by line during training.
+
+    Args:
+        data: A tf.data.Dataset, or a list of filenames.
+        vocabulary_size: The maximum size of a vocabulary to be trained.
+        vocabulary_output_file: The location to write a vocabulary file.
+        lowercase: If true, the input text will be first lowered before
+            tokenization.
+        strip_accents: If true, all accent marks will be removed from text
+            before tokenization.
+        split: If true, input will be split on whitespace and punctuation
+            marks, and all punctuation marks will be kept as tokens. If false,
+            input should be split ("pre-tokenized") before calling the
+            tokenizer, and passed as a dense or ragged tensor of whole words.
+            `split` is required to be`True` when `data` is a list of filenames.
+        suffix_indicator: The characters prepended to a wordpiece to indicate
+            that it is a suffix to another subword.
+        reserved_tokens: A list of tokens that must be included in the vocabulary.
+
+    Returns:
+        Returns a list of vocabulary terms.
+
+    Examples:
+
+    Basic Usage (from Dataset).
+    >>> inputs = tf.data.Dataset.from_tensor_slices(["bat sat pat mat rat"])
+    >>> vocab = compute_word_piece_vocabulary(inputs, 13)
+    >>> vocab
+    ['[PAD]', '[CLS]', '[SEP]', '[UNK]', '[MASK]', 'a', 'b', 'm', 'p', 'r', 's', 't', '##at']
+    >>> tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(vocabulary=vocab, oov_token="[UNK]")
+    >>> outputs = inputs.map(tokenizer.tokenize)
+    >>> for x in outputs: print(x)
```

Comment:
hmm is this format allowed `for x in outputs: print(x)`? Do we need the linebreak?

Reply:
Yep this format is allowed, the docstring compiles and is tested to print the same thing. However, if it is better style to add the linebreak, I can also add it